Docs updates for usage stats (#3393)
FusionAuth/fusionauth-site: astro/src/content/docs/get-started/core-concepts/collected-metrics.mdx

---
title: Collected Metrics
description: An overview of how, when, and why FusionAuth collects deployment usage data.
section: get started

> subcategory: download and install

Reviewer: Why is it in this subcategory? 

Should it be in both download and install and run in the cloud (like the setup wizard)?

Reviewer: Or maybe in the reference section? 

In the cloud, we collect these metrics but don't let people disable them. But we still should tell them what we collect.

Author: Add to both download and install, and run in the cloud. D&I should be the canonical. Look at Setup Wizard in first time setup. 

Change current Collected Metrics. We collect metrics. Network access. Community version can block. If you have a licensed version. If you want to learn more <link>

IF we do code version: it goes under docs_shared

---

> 

Reviewer: ```suggestion

## Overview

```

> One of our goals is constant improvement. In order to get there, we collect some usage data from FusionAuth deployments. This data helps us understand how you're using FusionAuth so that we can give you better features, tightened security, and tailored experiences.

Reviewer: ```suggestion
We college usage data from FusionAuth deployments to improve our product. This data helps us understand how you're using FusionAuth so that we can give you better features, tightened security, and tailored experiences.
```

> Currently FusionAuth sends a payload to the metrics endpoint for all instances.

Reviewer: ```suggestion
FusionAuth collects high level metrics for all instances.
```

> Collected metrics of this payload include:

Reviewer: Did you pull this out of the system requirements section? /docs/get-started/download-and-install/system-requirements#collected-metrics

If so, please update that section to point here.

-  a unique instance identifier
-  the total number of users
-  the total number of monthly active users
-  the total number of monthly logins
-  the number of nodes in the instance
-  the product version
-  the license Id (only for licensed instances with a plan)
-  the version of the metrics data (currently always version 1)

A self-hosted community user can disable network egress or block access to the metrics endpoint without affecting functionality.

Customers on a paid plan must allow a metrics connection [per the license agreement](/license), unless they have an air-gapped license.

> 

Reviewer: ```suggestion
### Detailed Metrics

<Aside type="version">
Available since 1.54.0
</Aside>
```

(you'll need to import the aside component as well)

> As of version 1.53.0, we have started collecting further usage data. Your trust means everything to us. We're committed to being up front about the data that we collect and how we use it. 

Reviewer: 1.54.0, not 1.53.0

Reviewer: "Your trust means everything to us" feels like a BS phrase. I'd cut it myself.

> First, what do we collect? Here's a sample JSON object:

Reviewer: ```suggestion
Here's a sample JSON object showing what we collect:
```

> <details>

Reviewer: Ooh, this is cool! I hadn't seen this component before.

  <summary>Usage Stats - Example JSON Payload</summary>
``` 
{

>   "applicationVersion": "1.53.0",

Reviewer: ```suggestion
  "applicationVersion": "1.54.0",
```

>   "collectionDuration": 14064,

Reviewer: Did you check that this is the latest version of a sample payload?

    "entitiesEntityGrantPermissions": 120,
    "entitiesEntityGrants": 40,
    "entitiesNumber": 120,
    "entitiesPermissions": 12,
    "entitiesTypeNumber": 4,
    "entitiesUserGrantPermissions": 120,
    "entitiesUserGrants": 40,
    "groupApplicationRoles": 60,
    "groupWithMembers": 30,
    "instanceDailyActiveUsers": 3300,
    "instanceDatabaseConnectionTimeout": 2000,
    "instanceDatabaseIdleTimeout": 120000,
    "instanceDatabaseMaxLifetime": 240000,
    "instanceDatabaseMaximumPoolSize": 10,
    "instanceDatabaseMinimumIdle": -1,
    "instanceDatabaseMysqlEnforceUtf8mb4": false,
    "instanceDatabaseVersion": "8.3.0",
    "instanceFirstTimeSetupProgress": {
      "apiKeyConfigured": false,
      "applicationConfigured": false,
      "emailConfigured": false,
      "licenseActivated": false
    },
    "instanceFusionauthAppHttpReadTimeout": 0,
    "instanceFusionauthAppLocalMetricsEnabled": false,
    "instanceFusionauthAppMemory": 25769803776,
    "instanceFusionauthAppReindexBatchSize": 2500,
    "instanceFusionauthAppReindexThreadCount": 5,
    "instanceFusionauthAppSearchDefaultRefreshInterval": "1s",
    "instanceFusionauthSearchHostsCount": 1,

>     "instanceJavaVersion": "21.0.2+13-LTS",

Reviewer: This is not a count.

    "webhookUserLoginSuspiciousTransactional": 5,
    "webhookUserPasswordBreach": 4,
    "webhookUserPasswordBreachTransactional": 6,
    "webhookUserPasswordResetSend": 4,
    "webhookUserPasswordResetStart": 4,
    "webhookUserPasswordResetSuccess": 4,
    "webhookUserPasswordUpdate": 4,
    "webhookUserReactivate": 5,
    "webhookUserReactivateTransactional": 6,
    "webhookUserRegistrationCreate": 4,
    "webhookUserRegistrationCreateComplete": 4,
    "webhookUserRegistrationCreateTransactional": 6,
    "webhookUserRegistrationDelete": 4,
    "webhookUserRegistrationDeleteComplete": 5,
    "webhookUserRegistrationDeleteTransactional": 6,
    "webhookUserRegistrationUpdate": 4,
    "webhookUserRegistrationUpdateComplete": 4,
    "webhookUserRegistrationUpdateTransactional": 6,
    "webhookUserRegistrationVerified": 4,
    "webhookUserRegistrationVerifiedTransactional": 6,
    "webhookUserTwoFactorMethodAdd": 3,
    "webhookUserTwoFactorMethodRemove": 5,
    "webhookUserUpdate": 4,
    "webhookUserUpdateComplete": 4,
    "webhookUserUpdateTransactional": 6
  }
}
```
</details>

You can see here that we are collecting information about feature usage.

> We will never collect anything personally identifiable. We may collect new metrics as we add new features, but those will be outlined in our release notes.

Reviewer: > We will never collect anything personally identifiable.

This is not correct. We capture instance Id, which can be tied to our customer. What we promise is that we'll never capture anything personally identifiable about our users users or customers.

> Here's the breakdown of who we will collect data from, and how it will happen:

Reviewer: ```suggestion
Here's the breakdown of what FusionAuth deployments which this feature will collect data from, and how it will happen:
```

Maybe?

> -   Self-hosted: You get to choose whether or not you want to share your data with us. You can turn usage data collection off by API or in the admin UI. You can also block the outbound request at the network--just block the hostname https://usage-stats.fusionauth.io. If you decide to share usage data, know that we'll handle your data with the highest level of care and security.

Reviewer: ```suggestion
-   Self-hosted: You get to choose whether or not you want to share your data with us. You can turn usage data collection off by API, setup wizard, or in the admin UI, under XXX. You can also block the outbound request at the network--just block the hostname https://usage-stats.fusionauth.io. If you decide to share usage data, we will protect it.
```

where you need to replace XXX with the path to the admin UI screen where they can turn it off.

Also suggested change about the protection to make it a bit less flowery :) .

-   By default, any upgraded instances will NOT send usage data.
-   By default, any new instances will send usage data, but you can opt out at instance creation or any time after.

> -   FA-Cloud: For those who choose to have us host your FusionAuth deployment, data collection is part of the package. It helps us give you top-notch service and security, and we promise that we're only using that data to make FusionAuth better for you.

Reviewer: ```suggestion
-   For those who choose to have us host your FusionAuth deployment in FusionAuth Cloud, data collection can not be disabled. This data helps us provide top-notch service and security.
```

> -   Community plan: If you're using an unlicensed version of FusionAuth, we'll only be collecting data points. If you do choose Licensed Community (available in version 1.52.0 and later, which adds free passkeys and other features), we will collect the same data, but tie it to your account and identity.

Reviewer: ```suggestion
- If you are on the Community plan, which all unlicensed versions of FusionAuth are, and enable this feature, we collect data but won't tie it to your identity. 
- If you are on the Licensed Community plan (available in version 1.52.0 and later, which enables passkeys and other features), we tie usage data to your account and identity.
```

> -   Other plans: We collect the same data and tie it to your account and identity.

Reviewer: ```suggestion
-   Other plans: We tie the data to your account and identity via the `instanceId` field.
```

> If you have questions or concerns, we're here to chat. Drop us an email at [devrel@fusionauth.io](mailto:devrel@fusionauth.io), or stop by the [FusionAuth Community forums](/community).

Reviewer: ```suggestion
If you have questions or concerns, we're here to chat. Drop us an email at [devrel@fusionauth.io](mailto:devrel@fusionauth.io), or stop by the [FusionAuth Community forums](/community/forum/).
```

> FAQ: 

Reviewer: ```suggestion
## Frequently Asked Questions

```

> -----

Reviewer: ```suggestion
```

> ### How will this impact performance? 

Reviewer: Is this the proper casing?

Author: Well, that depends. If we're using the Inversoft style guide then yes. Though if I'm not mistaken, don't we have an automation that changes everything to the correct case anyway?

Reviewer: How does it render if you spin up the local site?

We have been developing this feature internally for quite some time. First, stats collection is intentionally throttled to not use additional resources. We then chose to only have the queries run once per day. We load tested with a variety of customer sizes and database configurations until we were able to collect this data without the collection impacting deployment performance. As a further failsafe, we have built in functionality that lets us dynamically disable any stat collection that does cause a problem, without the need to release or upgrade the instance.

### How does FusionAuth ensure that my customer data remains private? 

Your privacy, and that of your customers, is of the utmost importance to us. We have designed this system so that it only collects aggregated data, so that we do not see data on any one specific instance. Further, we do not collect any data that might reveal any information about your customers, such as webhook or OIDC URLs.

### Who owns the data that you collect? Who decides how that data is used?

> The FusionAuth license agreement covers the specifics of how we collect and use data. For more detailed information, please see the following sections:

Reviewer: Would add link to the license agreement.

-   Section 2.1.6 covers self-hosting
-   Section 2.2.4 discusses those hosting with FusionAuth cloud
-   Section 5.2 covers FusionAuth's licensing of collected data
-   Section 5.3 has the information on consent for feedback

### How does your data collection impact our ability to remain GDPR or CCPA compliant?

We know how critical compliance is for your business. For many customers, it's a primary driver in why they chose FusionAuth. Because of this, we made the choice to avoid collecting any personally identifiable information. We collect all stats in aggregate, so that there is no chance of violating confidentiality or compliance.

### What steps are you taking to avoid scope creep and collecting ever-more data?

We're aware of how easy it is to let a project like this expand. Who doesn't like more data? Before we started this project, we decided on the following principles:

-   Self-hosting customers will always be able to turn off usage data collection.

> -   We limit data collection to counts around usage of the product.

Reviewer: ```suggestion
-   We limit data collection to feature usage counts and software versions of the product.
```

-   We will never collect anything that would let us know specific information about your configuration or your customers such as OIDC URLs, user emails or mail hosts. 
-   We will not collect new stats without releasing a new version of FusionAuth.

> Our thought process is two-fold: first, we want to do right by our customers. Second, by making these commitments, we are limiting target data. Our developers are busy, just like you and yours. It's in our shared best interest to be transparent about data collection, and to prevent scope creep where possible.

Reviewer: ```suggestion
We want to do right by our customers. By making these public commitments, we are clear about what data we gather and what data we will not. We want to be transparent about data collection and prevent scope creep.
```
